vm/package/deploy.rs

```diff
@@ -71,39 +78,47 @@ impl<'de, N: Network> Deserialize<'de> for DeployRequest<N> {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
         // Parse the request from a string into a value.
         let request = serde_json::Value::deserialize(deserializer)?;
-        // Recover the leaf.
-        Ok(Self::new(
+        // Build the deploy request.
+        let deploy_request = Self::new(
             // Retrieve the program.
-            serde_json::from_value(request["deployment"].clone()).map_err(de::Error::custom)?,
+            serde_json::from_value(request["transaction"].clone()).map_err(de::Error::custom)?,
             // Retrieve the address of the program.
             serde_json::from_value(request["address"].clone()).map_err(de::Error::custom)?,
             // Retrieve the program ID.
             serde_json::from_value(request["program_id"].clone()).map_err(de::Error::custom)?,
-        ))
+        );
+        match deploy_request {
+            Ok(deploy_request) => Ok(deploy_request),
+            Err(error) => Err(de::Error::custom(error.to_string())),
+        }
     }
 }
 
 pub struct DeployResponse<N: Network> {
-    deployment: Deployment<N>,
+    transaction: Transaction<N>,
 }
 
 impl<N: Network> DeployResponse<N> {
     /// Initializes a new deploy response.
-    pub const fn new(deployment: Deployment<N>) -> Self {
-        Self { deployment }
+    pub fn new(transaction: Transaction<N>) -> Result<Self> {
+        ensure!(
+            matches!(transaction, Transaction::Deploy(_, _, _)),
+            "Cannot create a deploy response with an execution transaction"
```

Comment:
```suggestion
            "Cannot create a deploy response with a non-deployment transaction"
```

Reply:
^We intend to support other transaction types in the future, hence the greater specificity.